samcli/lib/build/app_builder.py

```diff
@@ -277,8 +278,8 @@ def update_template(
         template_dict = stack.template_dict
 
         for logical_id, resource in template_dict.get("Resources", {}).items():
-
-            full_path = get_full_path(stack.stack_path, logical_id)
+            resource_id = ResourceMetadataNormalizer.get_resource_id(resource, logical_id)
+            full_path = get_full_path(stack.stack_path, resource_id)
```

Comment:
nit: let's keep using `logical_id` or `physical_id` for resource ids in the template.

Reply:
resource_id is not always logical Id, this is a new introduced id. In case of CDK (or generally any IaC), it will be the CDK id (or generally the IaC id), in case of CFN, it will be the resource Logical Id

Reply:
I agree, I renamed it to be `resource_iac_id`